libcperciva/datastruct/seqptrmap.c

```diff
@@ -79,12 +79,19 @@ seqptrmap_add(struct seqptrmap * M, void * ptr)
 void *
 seqptrmap_get(struct seqptrmap * M, int64_t i)
 {
-	size_t pos = i - M->offset;
+	size_t pos;
+	int64_t pos_i;
 
 	/* Sanity check, but we cannot fail, so don't use an assert. */
 	if (i < 0)
 		return (NULL);
 
+	/* Check that the position will fit into size_t. */
+	pos_i = i - M->offset;
+	if ((pos_i < 0) || ((uint64_t)pos_i >= (uint64_t)SIZE_MAX))
```

Comment:
if `size_t` was `uint128_t`, then I guess this line could cause problems?  i.e. the `(uint64_t)SIZE_MAX` cast would be implementation-defined.  If that meant "discard the high bits", then this would still work, but theoretically the implementation could define "nasal demons" in this case?